apis/metal3.io/v1alpha1/firmwareschema_types.go

```diff
@@ -0,0 +1,147 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// Additional data describing the firmware setting
+type SettingSchema struct {
+
+	// The type of setting.
+	// +kubebuilder:validation:Enum=Enumeration;String;Integer;Boolean;Password
```

Comment:
Do we expect to support non-Redfish implementation? If yes, we cannot add this validation.

Reply:
Yes, for non-redfish types there will be no BIOS registry data returned and therefore no schema, so no validations can be done on the settings. I will note this in the CRD.

Reply:
Well, other drivers may decide to implement schema as well.